### PR TITLE
refactor/98: Refactored propertyToXxxReaderClasses to use Lombok

### LINKED ISSUE #98: Refactor propertyToXxxReaderClasses to use Lombok
Refactor propertyToXxxReaderClasses to use Lombok to make the code less verbose

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #98

### CODE REVIEW

File: src/main/java/com/exasol/adapter/document/mapping/IterationIndexColumnMapping.java [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
  * This class defines a column that maps the array index of a nested list. Such columns are useful for nested tables
  * that do not have an key.
  */
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Data
+@SuperBuilder(toBuilder = true)
 public final class IterationIndexColumnMapping extends AbstractColumnMapping {
-    private static final long serialVersionUID = -2526873217496416853L;//
+    private static final long serialVersionUID = -5720702055496015560L;//
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/document/mapping/IterationIndexColumnMapping.java:18]:
```suggestion
    private static final long serialVersionUID = -5720702055496015560L;
```

Comment (morazow) [reply src/main/java/com/exasol/adapter/document/mapping/IterationIndexColumnMapping.java:18]: You have it some of them but not all, so please remove them to keep consistent if not necessary.

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/adapter/document/mapping/AbstractPropertyToColumnMapping.java [hunk 2]
```diff
@@ -1,75 +1,24 @@
...
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
 /**
  * This class is an abstract basis for {@link PropertyToColumnMapping}s.
  */
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Data
+@SuperBuilder(toBuilder = true)
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/mapping/AbstractPropertyToColumnMapping.java:14]: No it's not

File: src/main/java/com/exasol/adapter/document/mapping/PropertyToDecimalColumnMapping.java [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
  * This class defines a mapping that extracts a decimal number from the remote document and maps it to an Exasol DECIMAL
  * column.
  */
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Data
+@SuperBuilder(toBuilder = true)
 public final class PropertyToDecimalColumnMapping extends AbstractPropertyToColumnMapping {
-    private static final long serialVersionUID = 6021806680404016342L;//
+    private static final long serialVersionUID = 6_021_806_680_404_016_342L;//
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/mapping/PropertyToDecimalColumnMapping.java:17]:
```suggestion
    private static final long serialVersionUID = 6021806680404016342L;
```

**Review by morazow — APPROVED** [review]

**Pull request merged**